apps/common/js/components/leap-hands-view.jsx

```diff
@@ -56,19 +76,22 @@ export default class LeapHandsView extends React.Component {
     camera.position.fromArray([0, 500, 400]);
     camera.lookAt(new THREE.Vector3(0, 0, 0));
     scene.add(camera);
-    return {scene, camera};
+    return { scene, camera };
   }
 
   render() {
-    const { width, height } = this.props;
+    let width = this.width;
+    let height = this.height;
+    if (this.props.width != null){
+      width = this.props.width;
+      height = this.props.height;
+    }
```

Comment:
> needed to make sure that the containers were resized as well as the renderer viewport.

Right, I forgot about renderer resizing before when I mentioned that CSS should be enough! That makes it more complicated.

My typical pattern when I'm dealing with this issue is to change dimensions via CSS and then notify child components that dimensions have been updated. Actually, it's used by the GRASP Seasons project, as there's exactly the same situation - when user rearranges views, I need to update 3D renderers too. You might take a look at it. 
So, in this case it could look look like:
- `activeViewPanelChanged` callback triggers change of the container size via CSS (e.g. different class applied). It happens now.
- But, we also would need call there: `this.refs.instructionsOverlay.resize()` to notify it that container dimensions are updated (well, or .'notifyResize` or whatever sounds good).
- Then `this.refs.instructionsOverlay.resize` would need to call `this.refs.handsView.resize()`.
- Finally, `this.refs.handsView.resize` calls `this.renderer.setSize(...)` and `this.camera.setAspectRatio(...)`.

So, it's chain of notifications. Each component needs to know which children can have problem with dynamic dimensions sizes (in practice all the components that use 3D rendering) and let them know that it happened.
Not sure if it's the best solution, but that's what I've used before. 

>  the initial rendering call fixed the aspect ratio of the hands

I guess we need to save camera reference (`this.camera = ...` instead of `const camera = ...`) and then we could use it to update aspect ratio.

HTH